src/util.cc

```diff
@@ -423,21 +427,82 @@ const char* SpellcheckString(const char* text, ...) {
 }
 
 #ifdef _WIN32
+std::wstring Utf8ToWide(const std::string& u8_string) {
+  if (u8_string.empty())
+    return std::wstring();
+
+  size_t size_needed =
+      MultiByteToWideChar(CP_UTF8, 0, u8_string.data(),
+                          static_cast<int>(u8_string.size()), NULL, 0);
+  if (size_needed == 0)
+    Fatal("Failed conversion to wide string");
+
+  vector<wchar_t> w_buffer(size_needed);
+  int w_size_needed = MultiByteToWideChar(CP_UTF8, 0, u8_string.data(),
+                                          static_cast<int>(u8_string.size()),
+                                          &w_buffer[0], w_buffer.size());
+  if (w_size_needed == 0)
+    Fatal("Failed conversion to wide string");
+
+  return std::wstring(&w_buffer[0], w_size_needed);
+}
+
+std::string WideToUtf8(const std::wstring& w_string) {
+  if (w_string.empty())
+    return std::string();
+
+  int size_needed = WideCharToMultiByte(CP_UTF8, 0, &w_string[0],
+                                        static_cast<int>(w_string.size()), NULL,
+                                        0, NULL, NULL);
+  if (size_needed == 0)
+    Fatal("Failed conversion to UTF-8 string");
+
+  std::string u8_string(size_needed, 0);
+  WideCharToMultiByte(CP_UTF8, 0, &w_string[0],
```

Comment:
Assert size needed has not changed.

Also, you're not retrieving the return value, and instead comparing the previous size_needed (which you've already asserted)